bundles/org.openhab.binding.teslapowerwall/src/main/java/org/openhab/binding/teslapowerwall/internal/TeslaPowerwallWebTargets.java

```diff
@@ -0,0 +1,160 @@
+            throws TeslaPowerwallCommunicationException, TeslaPowerwallAuthenticationException {
+        String response = invoke(getBatterySOEUri, email, password);
+        logger.trace("getBatterySOE response = {}", response);
+        return BatterySOE.parse(response);
+    }
+
+    public GridStatus getGridStatus(String email, String password)
+            throws TeslaPowerwallCommunicationException, TeslaPowerwallAuthenticationException {
+        String response = invoke(getGridStatusUri, email, password);
+        logger.trace("getGridStatus response = {}", response);
+        return GridStatus.parse(response);
+    }
+
+    public SystemStatus getSystemStatus(String email, String password)
+            throws TeslaPowerwallCommunicationException, TeslaPowerwallAuthenticationException {
+        String response = invoke(getSystemStatusUri, email, password);
+        logger.trace("getSystemStatus response = {}", response);
+        return SystemStatus.parse(response);
+    }
+
+    public MeterAggregates getMeterAggregates(String email, String password)
+            throws TeslaPowerwallCommunicationException, TeslaPowerwallAuthenticationException {
+        String response = invoke(getMeterAggregatesUri, email, password);
+        logger.trace("getMeterAggregates response = {}", response);
+        return MeterAggregates.parse(response);
+    }
+
+    public Operations getOperations(String email, String password)
+            throws TeslaPowerwallCommunicationException, TeslaPowerwallAuthenticationException {
+        String response = invoke(getOperationUri, email, password);
+        logger.trace("getOperations response = {}", response);
+        return Operations.parse(response);
+    }
+
+    public String getToken(String email, String password)
+            throws TeslaPowerwallCommunicationException, TeslaPowerwallAuthenticationException {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty("username", "customer");
+        jsonObject.addProperty("password", password);
+        jsonObject.addProperty("email", email);
+        jsonObject.addProperty("force_sm_off", false);
+        logger.debug("logonjson = {}", jsonObject.toString());
+        String response = invoke(getTokenUri, HttpMethod.POST.asString(), "Content-Type", "application/json",
+                jsonObject.toString());
+        JsonObject jsonResponse = JsonParser.parseString(response).getAsJsonObject();
+        String token = jsonResponse.get("token").getAsString();
+        logger.debug("Token: {}", token);
+        return token;
+    }
+
+    private String invoke(String uri, String email, String password)
+            throws TeslaPowerwallCommunicationException, TeslaPowerwallAuthenticationException {
+        if (token.isEmpty()) {
+            token = getToken(email, password);
+        }
+        return invoke(uri, HttpMethod.GET.asString(), "Authorization", "Bearer " + token, "");
```

Comment:
```suggestion
        return invoke(uri, HttpMethod.GET, "Authorization", "Bearer " + token, "");
```

This should be coombined with the opther comment about the method signature.

Reply:
Done.